ogre2/src/Ogre2InertiaVisual.cc

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "ignition/rendering/ogre2/Ogre2InertiaVisual.hh"
+#include "ignition/rendering/ogre2/Ogre2Material.hh"
+#include "ignition/rendering/ogre2/Ogre2DynamicRenderable.hh"
+
+#ifdef _MSC_VER
+  #pragma warning(push, 0)
+#endif
+#include <OgreSceneNode.h>
+#ifdef _MSC_VER
+  #pragma warning(pop)
+#endif
+
+using namespace ignition;
+using namespace rendering;
+
+class ignition::rendering::Ogre2InertiaVisualPrivate
+{
+  /// \brief inertia visual materal
+  public: Ogre2MaterialPtr material = nullptr;
+
+  /// \brief Ogre renderable used to render the cross lines.
+  public: std::shared_ptr<Ogre2DynamicRenderable> crossLines = nullptr;
+
+  /// \brief Box visual
+  public: VisualPtr boxVis = nullptr;
+};
+
+//////////////////////////////////////////////////
+Ogre2InertiaVisual::Ogre2InertiaVisual()
+  : dataPtr(new Ogre2InertiaVisualPrivate)
+{
+}
+
+//////////////////////////////////////////////////
+Ogre2InertiaVisual::~Ogre2InertiaVisual()
+{
+}
+
+//////////////////////////////////////////////////
+void Ogre2InertiaVisual::PreRender()
+{
+}
+
+//////////////////////////////////////////////////
+void Ogre2InertiaVisual::Init()
+{
+  BaseInertiaVisual::Init();
+}
+
+//////////////////////////////////////////////////
+void Ogre2InertiaVisual::Destroy()
+{
+  if (this->dataPtr->boxVis != nullptr)
+  {
+    this->dataPtr->boxVis->Destroy();
```

Comment:
reset `boxVis` so when `Destroy` is called again, the above line won't be executed.

```
this->dataPtr->boxVis.reset();
```

After that, apply this patch, and see if it fixes the crash for you

```
diff --git a/ogre2/src/Ogre2Node.cc b/ogre2/src/Ogre2Node.cc
index 63a43a93..cff1e3e8 100644
--- a/ogre2/src/Ogre2Node.cc
+++ b/ogre2/src/Ogre2Node.cc
@@ -64,6 +64,8 @@ Ogre::SceneNode *Ogre2Node::Node() const
 //////////////////////////////////////////////////
 void Ogre2Node::Destroy()
 {
+  if (ogreNode)
+    return;
```

Reply:
The crash is fixed, but `UNIT_Scene_TEST` seems to be failing

```
      Start 55: UNIT_Scene_TEST
55/93 Test #55: UNIT_Scene_TEST .......................***Failed    1.25 sec

```

Reply:
oops that should've been `if (!ogreNode)`

Can you take a look at the changes in this commit:

https://github.com/ignitionrobotics/ign-rendering/commit/9575d3ddc0f4524348a4b203e6b1e3a51e33c6a3

I think that fixes the test and crash.

Reply:
Fixed in [`dd42817`](https://github.com/ignitionrobotics/ign-rendering/pull/326/commits/dd42817dd2b396b738ff0988850e433f41f1cd4a)